packages/integration-sdk-core/src/types/step.ts

```diff
@@ -113,6 +113,11 @@ export type IntegrationStepResult = Omit<
    * Duration of the step in milliseconds, calculated as endTime - startTime.
    */
   duration?: number;
+  /**
+   * TODO: Decouple steps from uploads. This shouldn't be here.
+   * This are types that are involved in a failed upload. We should mark them as partial.
+   */
+  partialTypesToForce?: string[];
```

Comment:
I def don't like to have this information in the step result, but might be the only way.